src/components/POLine/Item/ItemForm.js

```diff
@@ -31,25 +40,50 @@ class ItemForm extends Component {
     stripes: PropTypes.object.isRequired,
     dispatch: PropTypes.func.isRequired,
     change: PropTypes.func.isRequired,
+    identifierTypes: PropTypes.arrayOf(PropTypes.object),
+    initialValues: PropTypes.object,
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      instanceId: '',
+      title: '',
+      publisher: '',
+      publicationDate: '',
+      edition: '',
+      contributors: [],
+      productIds: [],
+    };
+  }
+
   onAddInstance = (instance) => {
-    const { dispatch, change } = this.props;
-    const { contributors, editions, publication, title } = instance;
+    const { dispatch, change, identifierTypes } = this.props;
+    const { contributors, editions, publication, title, identifiers, id } = instance;
 
-    dispatch(change('title', title));
+    dispatch(change('instanceId', id));
+    this.setState({ instanceId: id });
+    if (title) {
+      dispatch(change('title', title));
+      this.setState({ title });
+    }
     if (publication && publication.length) {
       const { publisher, dateOfPublication = '' } = publication[0];
 
       dispatch(change('publisher', publisher));
+      this.setState(({ publisher }));
```

Comment:
I think it could be done with one `this.setState`

Reply:
Ok. Done